backends/bmv2/common/deparser.cpp

```diff
@@ -24,14 +24,53 @@ void DeparserConverter::convertDeparserBody(const IR::Vector<IR::StatOrDecl>* bo
                                           Util::JsonArray* order, Util::JsonArray* primitives) {
     ctxt->conv->simpleExpressionsOnly = true;
     for (auto s : *body) {
+        auto isR = false;
+        IR::MethodCallExpression *mce2 = nullptr;
         if (auto block = s->to<IR::BlockStatement>()) {
             convertDeparserBody(&block->components, order, primitives);
             continue;
         } else if (s->is<IR::ReturnStatement>() || s->is<IR::ExitStatement>()) {
             break;
         } else if (s->is<IR::EmptyStatement>()) {
             continue;
-        } else if (s->is<IR::MethodCallStatement>()) {
+        } else if (s->is<IR::AssignmentStatement>()) {
+            auto assign = s->to<IR::AssignmentStatement>();
+            if (assign->right->is<IR::MethodCallExpression>()) {
+                auto mce = assign->right->to<IR::MethodCallExpression>();
+                auto minst = P4::MethodInstance::resolve(mce, ctxt->refMap, ctxt->typeMap);
+                if (minst->is<P4::ExternMethod>()) {
+                    auto extmeth = minst->to<P4::ExternMethod>();
+                if ((extmeth->method->name.name == "get"
+                    || extmeth->method->name.name == "get_state")
+                    && extmeth->originalExternType->name == "InternetChecksum") {
+                        const IR::Expression *l;
+                        l = assign->left;
+                        isR = true;
+                        auto dest = new IR::Argument(l);
+                        auto args = new IR::Vector<IR::Argument>();
+                        args->push_back(dest);  // dest
+                        mce2 = new IR::MethodCallExpression(mce->method, mce->typeArguments);
+                        mce2->arguments = args;
+                        s = new IR::MethodCallStatement(mce);
+                    }
+                }
+            } else if (assign->right->is<IR::Constant>()) {
+                ctxt->conv->simpleExpressionsOnly = false;
+                auto json = new Util::JsonObject();
+                auto params = mkArrayField(json, "parameters");
+                auto type = ctxt->typeMap->getType(assign->left, true);
+                json->emplace("op", "set");
+                auto l = ctxt->conv->convertLeftValue(assign->left);
+                bool convertBool = type->is<IR::Type_Boolean>();
+                auto r = ctxt->conv->convert(assign->right, true, true, convertBool);
+                params->append(l);
+                params->append(r);
+                primitives->append(json);
+                ctxt->conv->simpleExpressionsOnly = true;
+                continue;
+            }
```

Comment:
shouldn't there be an else clause here that gives an error for other kinds of expressions on the right?

Reply:
I will add it.